src/Promise.js

```diff
@@ -210,6 +210,27 @@ Promise.prototype.always = function (fn) {
   return this.then(fn, fn);
 };
 
+/**
+  * Execute given callback when the promise either resolves or rejects.
+  * Same semantics as Node's Promise.finally()
+  * @param {Function} fn
+  * @returns {Promise} promise
+  */
+Promise.prototype.finally = function (fn) {
+  var me = this;
+  const res = function() {
+    return me;
+  }
+  const final = function() {
+    const wrapper = Promise.defer();
+    wrapper.promise.then(res);
```

Comment:
Actually, what does this line do? Is there some side effect?

Reply:
This chains the current Promise as the return from the newly created promise as to keep the promise chain in tact. Without this a call to finally would result in the promise chain ending.

Reply:
Shouldn't that then be `wrapper.promise = wrapper.promise.then(res)`?

Reply:
Good point! I went ahead and updated the imlpementation to no longer use `Promise.defer` and to just new up a `Promise` instance as it simplified the implementation since we always `resolve` the promise.

I also removed the test checking for a value returned from `finally` as I was mistaken and the spec does not specify this.